test/integration/targets/vmware_vmotion/tasks/main.yml

```diff
@@ -0,0 +1,79 @@
+# Test code for the vmware_vmotion module.
+# Copyright: (c) 2019, Pavan Bidkar <pbidkar@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- import_role:
+    name: prepare_vmware_tests
+  vars:
+    setup_attach_host: true
+    setup_virtualmachines: True
+    setup_datastore: true
+
+- name: Create VMs
+  vmware_guest:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    datacenter: "{{ dc1 }}"
+    validate_certs: no
+    name: test_vm1
+    folder: vm
+    state: present
+    guest_id: centos7_64Guest
+    disk:
+    - size_gb: 1
+      type: thin
+      datastore: '{{ rw_datastore }}'
+    hardware:
+      version: latest
+      memory_mb: 1024
+      num_cpus: 1
+      scsi: paravirtual
+  register: vm_create
+
+- name: Perform vMotion of virtual machine
+  vmware_vmotion:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    validate_certs: no
+    vm_name: test_vm1
+    destination_host: '{{ esxi2 }}'
+  register: vm_vmotion
```

Comment:
The step fails with the following error:

```
TASK [vmware_vmotion : Perform vMotion of virtual machine to resource_pool] ******************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: IndexError: list index out of range
fatal: [testhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/home/centos/.ansible/tmp/ansible-tmp-1574258752.4017994-249217622119947/AnsiballZ_vmware_vmotion.py\", line 116, in <module>\n    _ansiballz_main()\n  File \"/home/centos/.ansible/tmp/ansible-tmp-1574258752.4017994-249217622119947/AnsiballZ_vmware_vmotion.py\", line 108, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/centos/.ansible/tmp/ansible-tmp-1574258752.4017994-249217622119947/AnsiballZ_vmware_vmotion.py\", line 54, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.cloud.vmware.vmware_vmotion', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_vmware_vmotion_payload_iigldg5s/ansible_vmware_vmotion_payload.zip/ansible/modules/cloud/vmware/vmware_vmotion.py\", line 374, in <module>\n  File \"/tmp/ansible_vmware_vmotion_payload_iigldg5s/ansible_vmware_vmotion_payload.zip/ansible/modules/cloud/vmware/vmware_vmotion.py\", line 370, in main\n  File \"/tmp/ansible_vmware_vmotion_payload_iigldg5s/ansible_vmware_vmotion_payload.zip/ansible/modules/cloud/vmware/vmware_vmotion.py\", line 172, in __init__\n  File \"/tmp/ansible_vmware_vmotion_payload_iigldg5s/ansible_vmware_vmotion_payload.zip/ansible/modules/cloud/vmware/vmware_vmotion.py\", line 341, in get_vm\nIndexError: list index out of range\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

Reply:
Actually this was the reason I added setup_virtualmachine=True step. I came across this error.
We get this error if there are no VM in testbed. This is existing code. to find the VM. 

But I dont see it my test runs

Reply:
Actually Error message returned here is not correct. module should Exit with Failure message VM not found 
I will create separate issue and PR for this fix